src/Dapr.Actors/Runtime/ActorManager.cs

```diff
@@ -185,18 +186,35 @@ async Task<byte[]> RequestFunc(Actor actor, CancellationToken ct)
             }
         }
 
-        internal Task FireTimerAsync(ActorId actorId, string timerName, CancellationToken cancellationToken = default)
+        internal async Task FireTimerAsync(ActorId actorId, string timerName, Stream requestBodyStream, CancellationToken cancellationToken = default)
         {
+            var timerData = await TimerInfo.DeserializeAsync(requestBodyStream);
+
+            dynamic awaitable;
             // Create a Func to be invoked by common method.
             async Task<byte[]> RequestFunc(Actor actor, CancellationToken ct)
             {
-                await
-                    actor.FireTimerAsync(timerName);
+                var actorTypeName = this.actorService.ActorTypeInfo.ActorTypeName;
+                var actorType = this.actorService.ActorTypeInfo.ImplementationType;
+                MethodInfo[] methods = actorType.GetMethods(BindingFlags.Public | BindingFlags.Instance);
+                var methodInfo = actorType.GetMethod(timerData.Callback);
+
+                var parameters = methodInfo.GetParameters();
+                if (parameters.Length == 0)
+                {
+                    awaitable = methodInfo.Invoke(actor, null);
+                }
+                else
+                {
+                    var deserializedType = timerData.Data;
+                    awaitable = methodInfo.Invoke(actor, new object[] { deserializedType });
```

Comment:
Note, that this will blow up if the method returns void or anything other than `Task`.

I see a few options:
- Require the method to return `Task` and throw an exception explaining that it's required if it doesn't.
- Copy in https://github.com/dotnet/aspnetcore/tree/master/src/Shared/ObjectMethodExecutor and attribute it correctly (this code is Apache licensed, and the we need to retain the copyright info)
- Reverse engineer from https://github.com/dotnet/aspnetcore/blob/master/src/Shared/ObjectMethodExecutor/ObjectMethodExecutor.cs based on the 1-2 cases we need to handle (complicated)

I'd choose option 1.

Reply:
Will go with option 1